src/bindings/pygaia/scripts/classification/generate_params_report.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2006-2019  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Gaia
+#
+# Gaia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+import os.path, csv, argparse
+from gaia2.fastyaml import yaml
+from get_classification_results import ClassificationResults
+from optparse import OptionParser
+
+
+def generateParamsReport(project_file):
+    project = yaml.load(open(project_file, 'r'))
+    results_dir = project['resultsDirectory']
+    csv_file = project_file.rstrip('project') + 'report.tsv'
+
+    if os.path.exists(results_dir):
+        classifier_type = None # all types
+
+        cr = ClassificationResults()
+        print('Loading all results...')
+        cr.readResults(results_dir)
+
+        n_results = len(cr.results)
+        results = cr.best(n_results, classifier_type)
+        # results is a list of tuples sorted by accuracy with these fields:
+        # (accuracy, std, norm_Accuracy, norm_std, results, results_file)
+
+        # use position as index and sort by norm_accuracy
+        r = [(v[2], i) for i, v in enumerate(results)]
+        r = sorted(r)[::-1]
+        _, idx = zip(*r)
+
+        fieldnames = ['accuracy_rank',
+                      'accuracy',
+                      'accuracy_std',
+                      'normalized_accuracy_rank',
+                      'normalized_accuracy',
+                      'normalized_accuracy_std',
+                      'results_file',
+                      'param_file',
+                      'kernel',
+                      'C',
+                      'preprocessing',
+                      'balance_classes',
+                      'type',
+                      'classifier',
+                      'gamma',
+                      'evaluation']
+
+        results = [{fieldnames[0]: k + 1,  # accuracy_rank
```

Comment:
do you do this in order to be able to pass the fieldname list to the dictwriter? It's a bit unfortunate because there's a lot of duplicated data here, and reading `fieldnames[0]` etc is quite difficult to follow. I'm not sure of a better way to do it, other than duplicating the dictionary keys as text here.